shared/config/resolver.js

```diff
@@ -0,0 +1,50 @@
+const { dirname, join } = require('path')
+
+// Repository root directory
+const paths = require('./paths')
+
+/**
+ * Default lookup resolver paths
+ *
+ * @param {string} basePath - Base directory, for example `/path/to/package`
+ * @returns {{ modules: string[], stylesheets: string[], views: string[] }} Resolver paths
+ */
+module.exports = (basePath) => {
```

Comment:
**question** Do I understand correctly that the need for that resolver function is because the `require` that we use to resolve in packageEntryToPath would resolve things from the point of view of the `shared/lib` workspace (and as such miss the local install of `govuk-frontend` in favour of the the local one hoisted at root)?

**suggestion** If that's the case, would it be less code for us to maintain (if at all possible) to have `packageEntryToPath` receive the "local" `require` function from the workspace it's being ran in? I'm a bit worried about us having to keep an eye on that list of paths?

Reply:
Creating a `require` sounds a good solution to help keep the calls to `packageEntryToPath` closer to where they're used. Actually, all that function needs is the path from `import.meta.url` isn't it, it can centralise that `createRequire` all, maybe? 🤔

Reply:
Yeah that can work just for Node.js

Shall I remove all the changes for Sass and Nunjucks and just pass `import.meta.url` (ES) or `__filename` (CommonJS) to every `packageEntryToPath()` call that needs a local `govuk-frontend`?

I picked directories (versus module paths) so we have something that works for Rollup, Sass and Nunjucks

So we'll still need directories for the `@rollup/plugin-node-resolve` plugin:

```mjs
plugins: [
  resolve({
    modulePaths: ['???']
  })
]
```

Reply:
Removed, have a look at https://github.com/alphagov/govuk-frontend/pull/3757#discussion_r1224505955